src/useDependenciesLoaded.tsx

```diff
@@ -0,0 +1,123 @@
+import Olm from "@matrix-org/olm";
+import olmWasmPath from "@matrix-org/olm/olm.wasm?url";
+import { Integrations } from "@sentry/tracing";
+import i18n from "i18next";
+import { initReactI18next } from "react-i18next";
+import LanguageDetector from "i18next-browser-languagedetector";
+import Backend from "i18next-http-backend";
+import { useEffect, useState } from "react";
+import * as Sentry from "@sentry/react";
+
+import { getUrlParams } from "./UrlParams";
+import { Config } from "./config/Config";
+import { DEFAULT_CONFIG } from "./config/ConfigOptions";
+
+enum LoadState {
+  None,
+  Loading,
+  Loaded,
+}
+
+// We need to synchonysly track the if init on the deps is already called:
+// otherwise strict mode will trigger the effect twice with xxxState still beeing LoadState.None
+// resulting in calling init twice.
+let alreadyInitializedOlm = false;
+let alreadyInitializedConfig = false;
+let alreadyInitializedSentry = false;
+let alreadyInitializedi18n = false;
+
+export function useDependenciesLoaded(): boolean {
```

Comment:
Do we take the `alreadyInitialized...` vars into account here? Perhaps this could be simplified further by using a singleton store to which the hook would only subscribe to

(when suggesting using a custom hook I did not realize we would need to use the global vars - sorry I overlooked that)

Reply:
Alternatively, I would do `useState(alreadyInitializedX ? LoadState.Loaded : LoadState.None)`, I think that should work?